[Chao Yung] iP

chao890 opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen source](https://dansilvestre.com/productivity-quotes/) 

## Capabilities

Duke is here to help you manage your day to day events and tasks.

- Keeping track of your tasks
- Keeping track of deadlines
- Storing and deleting tasks
- Greeting you

## Setup

The installation and setup is easy. All you need to do is this:
1. Download the latest jar file from [here](https://github.com/chao890/ip/releases)
2. Run the jar file with the following command
    
bash
    java -jar /path/to/jar/file
    
3. You're all ready to use Duke 🔥 

## Roadmap

- [X] Basic Task Manager Functions
- [ ] Generative Chatting Functionality
- [ ] Conquer the World World Peace

_Excerpt from Duke source code_

```java
public class Duke {
    /**
     * Initializes and runs an instance of Duke.
     *
     * @param args Additional arguments.
     */
    public static void main(String[] args) {
        Duke shell = new Duke();
        shell.run();
    }
} ```

Commit:
Add Gradle support

chao890 committed:
Level-1

chao890 committed:
Level-2

chao890 committed:
Level-3

chao890 committed:
Level-3

chao890 committed:
Level-4

chao890 committed:
A-TextUiTesting

chao890 committed:
Level-5

chao890 committed:
Level-6

chao890 committed:
branch-level-7

chao890 committed:
branch-Level-8

chao890 committed:
branch-Level-8 part 2

chao890 committed:
ui rename

chao890 committed:
Merge tag 'branch-Level-8'

chao890 committed:
Merge tag 'branch-Level-7'

chao890 committed:
A-MoreOOP, A-Packages

chao890 committed:
branch-A-JavaDo

chao890 committed:
branch-A-JavaDo

chao890 committed:
Ui rename

chao890 committed:
Ui rename 2

chao890 committed:
Level-9

chao890 committed:
Merge tag 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Duke/Duke.java
#	src/main/java/Duke/Handler.java

chao890 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/tasks.txt

chao890 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

chao890 committed:
A-Gradle

chao890 committed:
A-JUnit

chao890 committed:
A-Jar

chao890 committed:
A-CheckStyle

kianseong commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate timeobject;
```
Could this name be more descriptive?
```suggestion
    protected LocalDate dueDate;
```

kianseong commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class Event extends Task {
+
+    LocalDate timeobject;
```
Could this variable name be more descriptive?

kianseong commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,74 @@
+package Duke;
+
+import java.io.*;
```
Perhaps we could only import the required classes?

kianseong commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                    cur = new ToDo(event.substring(4));
+                    break;
+                case "D":
+                    int indexofdeadline = event.lastIndexOf("|") + 1;
```
Should we use camel case for variable names?
```suggestion
                    int indexOfDeadline = event.lastIndexOf("|") + 1;
```

kianseong commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,44 @@
+        this.isDone = true;
+    }
+
+    public void UnmarkAsDone() {
```
Could this follow camelCase?

kianseong reviewed (commented):
LGTM! Maybe could work on naming your variables and methods, but other than that I like how you separated your classes.

chao890 committed:
Level-10

chao890 committed:
A-CheckStyle add ons

JonLamy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public String storeToString() {
```
Perhaps a JavaDoc header for others to understand this method more?

JonLamy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,117 @@
+    public String getResponse(String input) {
+        try {
+            Handler handler = new Handler(tasklist, ui);
+            if (input.equals("bye")) {
+                storage.save();
+                return ui.bye();
+            } else if (input.equals("list")) {
+                storage.save();
+                return ui.showList();
+            } else if (input.startsWith("done")) {
+                storage.save();
+                return handler.handleMark(input);
+            } else if (input.startsWith("todo")) {
+                storage.save();
+                return handler.handleToDo(input);
+            } else if (input.startsWith("deadline")) {
+                storage.save();
+                return handler.handleDeadline(input);
+            } else if (input.startsWith("event")) {
+                storage.save();
+                return handler.handleEvent(input);
+            } else if (input.startsWith("delete")) {
+                storage.save();
+                return handler.handleDelete(input);
+            } else if (input.startsWith("find")) {
+                storage.save();
+                return handler.handleFind(input);
+            } else {
+                throw new DukeUnknownTaskException();
+            }
```
Would it be more readable if a switch statement were used here?

JonLamy commented on src/main/java/duke/Handler.java:
```diff
@@ -0,0 +1,129 @@
+    public String addTask(Task task) {
+        taskList.add(task);
+        return ui.printTask(task);
+    }
```
I like the consistency of method naming and organization.

JonLamy reviewed (commented):
Looks great, well organized and easy to read! Maybe package tasks together for better source file organization? Very cool 👍

chao890 committed:
Add greet funcionality

Added greet chatbox when user first starts the application.

This prompts the user to start talking to the chatbot.

chao890 committed:
Added assertions

chao890 committed:
A-CodeQuality

chao890 committed:
Merge pull request #1 from chao890/branch-A-Assertions

Added assertions

chao890 committed:
Merge branch 'master' into branch-A-CodeQuality

chao890 committed:
Merge pull request #2 from chao890/branch-A-CodeQuality

A-CodeQuality

chao890 committed:
BCD-Extension

chao890 committed:
Removed unused methods

chao890 committed:
Fix merge conflict

chao890 committed:
Merge branch 'branch-Level-8'

chao890 committed:
Added User Guide
